pkg/kubeapply/kubeapply.go

```diff
@@ -110,63 +110,56 @@ func (collection YAMLCollection) ApplyAndWait(
 		deadline := time.Now().Add(perPhaseTimeout)
 		err := applyAndWait(ctx, kubeinfo, deadline, debug, dryRun, collection[phaseName])
 		if err != nil {
-			if err == errorDeadlineExceeded {
-				err = errors.Errorf("phase %q not ready after %v", phaseName, perPhaseTimeout)
+			if errors.Is(err, errorDeadlineExceeded) {
+				err = fmt.Errorf("phase %q not ready after %v: %w", phaseName, perPhaseTimeout, err)
 			}
 			return err
 		}
 	}
 	return nil
 }
 
-func applyAndWait(ctx context.Context, kubeinfo *k8s.KubeInfo, deadline time.Time, debug, dryRun bool, filenames []string) error {
-	expanded, err := expand(ctx, filenames)
+func applyAndWait(ctx context.Context, kubeinfo *k8s.KubeInfo, deadline time.Time, debug, dryRun bool, sourceFilenames []string) error {
+	expandedFilenames, err := expand(ctx, sourceFilenames)
 	if err != nil {
-		return err
+		return fmt.Errorf("expanding YAML: %w", err)
 	}
 
 	cli, err := k8s.NewClient(kubeinfo)
 	if err != nil {
-		return errors.Wrapf(err, "kubeapply: error connecting to cluster %v", kubeinfo)
+		return fmt.Errorf("connecting to cluster %v: %w", kubeinfo, err)
 	}
 	waiter, err := NewWaiter(cli.Watcher())
 	if err != nil {
 		return err
 	}
 
 	valid := make(map[string]bool)
-	var msgs []string
-	for _, n := range expanded {
-		err := waiter.Scan(ctx, n)
-		if err != nil {
-			msgs = append(msgs, fmt.Sprintf("%s: %v\n", n, err))
-			valid[n] = false
-		} else {
-			valid[n] = true
+	var scanErrs derror.MultiError
+	for _, filename := range expandedFilenames {
+		valid[filename] = true
+		if err := waiter.Scan(ctx, filename); err != nil {
+			scanErrs = append(scanErrs, fmt.Errorf("watch %q: %w", filename, err))
```

Comment:
Hmmm. It's a little jarring to see that `scanErrs` is a `MultiError`, then see the `append` call screaming "it's an array!" Could we maybe have `MultiError.AddError` or the like, to avoid exposing the implementation?

Reply:
It is an array! A named type can be anything, not just a struct. Wrapping it in a struct and making it less convenient seems wrong; KISS!

Compare to [`k8s.io/apimachinery/pkg/util/errors.Aggregate`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/errors#NewAggregate) which does avoid exposing the implementation.  In a comment in https://github.com/datawire/dlib/pull/31 one of the things I dinged it for is "Isn't a simple []error type, and so is clunky to work with."  The one thing I think the opaque-wrapping approach has going for it is that it can (and apimachinery does) make it impossible to have an invalid `"(0 errors; BUG: this should not be reported as an error)"` error.